src/jit/codegenlinear.cpp

```diff
@@ -896,10 +896,15 @@ void CodeGen::genUnspillRegIfNeeded(GenTree* tree)
                 inst_RV_TT(ins_Load(treeType, compiler->isSIMDTypeLocalAligned(lcl->gtLclNum)), dstReg, unspillTree);
             }
 #elif defined(_TARGET_ARM64_)
-            var_types   targetType = unspillTree->gtType;
-            instruction ins        = ins_Load(targetType, compiler->isSIMDTypeLocalAligned(lcl->gtLclNum));
-            emitAttr    attr       = emitTypeSize(targetType);
-            emitter*    emit       = getEmitter();
+            var_types targetType = unspillTree->gtType;
+            if (targetType != genActualType(varDsc->lvType) && !varTypeIsGC(targetType) && !varDsc->lvNormalizeOnLoad())
```

Comment:
Could you add a comment to describe the case where the type of the tree and `genActualType(varDsc->lvType)` don't match?

Reply:
@CarolEidt The comment would be a duplicate of the XARCH comment above.  Do you just want the XARCH comment moved out of the ifdef?

Reply:
Ah, I missed that. Actually, looking at the code above, I see the `// TODO-Cleanup: The following code could probably be further merged and cleaned up.` which, along with the xarch comment, probably covers it adequately. Thanks